### PR TITLE
feat/new transformation to add ids to elements in files

### PR DESCRIPTION
**Pull request by corymosiman12**
This is to support the BuildingSync schema PR https://github.com/BuildingSync/schema/pull/193

### CODE REVIEW

File: add-required-ids/main.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+print(type(elements_requiring_ids))
+print(type(elements_requiring_ids[0]))
+
+# Setup parse
+parser = etree.XMLParser(remove_blank_text=True)
+etree.set_default_parser(parser)
+etree.register_namespace('auc', BUILDINGSYNC_URI)
+
+# tree = etree.parse(files[0])
+# el0 = elements_requiring_ids[0]
```
Comment (macintoshpie) [inline add-required-ids/main.py]: Can we delete this?

File: add-required-ids/main.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    :param elements_requiring_ids: [xmlschema.validators.elements.XsdElement] Schema elements that have ID attributes
+    :return:
+    """
+    tree = etree.parse(file_name)
+    for el in elements_requiring_ids:
+        # A nice method to return the full path to an element, so as to avoid
+        # finding elements such as auc:LinkedPremises/auc:Building
+        xp = "//" + el.get_path().replace(f"{{{BUILDINGSYNC_URI}}}", 'auc:')
+        elements_in_file = tree.xpath(xp, namespaces=NAMESPACES)
+        if len(elements_in_file) > 0:
```
Comment (macintoshpie) [inline add-required-ids/main.py]: Can remove this line probably

**Review by macintoshpie — APPROVED** [review]
This all looks good to me, the comments aren't really that important, just thought I'd mention them.

However I do wonder if we should maybe make this a bit more generic, which would allow buildingsync users to use this as well. I'm thinking we just accept a path to file or directory as the first argument. for example
```python
if len(sys.argv) < 3:
    # download the schema if user doesn't provide path to one
    schema_path = "BuildingSync.xsd"
    urllib.urlretrieve(bsync_schema_uri, schema_path)
else:
    schema_path = sys.argv[2]

source = sys.argv[1]
files = [source]
if os.path.isdir(source):
    files = [os.path.join(source, f) for f in os.listdir(source) if f.endswith('.xml')]
# use schema_path and the files list
...
```

**Comment by corymosiman12** [comment]
@macintoshpie didn't want to have to require users to point to specific XSD url on Github, as that can get ugly.  Required schema directory to exist and for user to pass directory as argument

**Review by nllong — APPROVED** [review]
looks good!

**Comment by nllong** [comment]
any reason not to merge this down? @corymosiman12

**Pull request merged**